### PR TITLE
Modernize components

### PR DESCRIPTION
**Pull request by Xolsk**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
     "dependencies": {
         "@storybook/cli": "^7.4.6",
         "@xmldom/xmldom": "^0.8.10",
+        "moment": "^2.30.1",
```
Comment (gkhristian) [inline package.json:56]: Could we check if we can migrate to date.js?

File: src/lib/components/FieldGroup/FieldGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+import { withTheme } from 'styled-components';
+import Icon from '../Icon/Icon';
+
+import { Tooltip } from '../Tooltip';
+import withDataId from '@/components/DataId/withDataId';
+import { ButtonSize } from '../Button';
+import { StyledFieldGroup } from '@/styles/Fieldgroup/StyledFieldGroup';
+
+const propTypes = {
+    /**
```
Comment (gkhristian) [inline src/lib/components/FieldGroup/FieldGroup.tsx:18]: I would say these propTypes are not used anymore.

File: src/lib/components/Toaster/ToasterContext.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+type ToasterContextType = {
+    toast: ({ text, type }: { text: any; type: any }) => void;
```
Comment (gkhristian) [inline src/lib/components/Toaster/ToasterContext.tsx:4]: Not sure about type, but text could be an string

File: src/lib/styles/Table/StyledTable.ts
Comment (gkhristian) [inline src/lib/styles/Table/StyledTable.ts:1]: Review if it still necessary, I rememeber that this was done becouse limitations, now with the last version might not be necessary

File: tests/setup.ts [hunk 4]
```diff
@@ -1,10 +1,25 @@
 import { expect, afterEach } from 'vitest';
 import { cleanup } from '@testing-library/react';
 import matchers from '@testing-library/jest-dom/matchers';
+import { vi } from 'vitest';
 
 // extends Vitest's expect method with methods from react-testing-library
 expect.extend(matchers);
 
+Object.defineProperty(window, 'matchMedia', {
```
Comment (gkhristian) [inline tests/setup.ts:9]: Just out of curiosity, why it is needed?

Comment (Xolsk) [reply tests/setup.ts:9]: !! This part was not done by me, it came with the recovery of the Pagination component

Comment (gkhristian) [reply tests/setup.ts:9]: OKs, I will review it

**Pull request merged**